rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/common/event/RMInitialState.java

```diff
@@ -61,99 +57,87 @@
  * @since ProActive Scheduling 0.9
  */
 @PublicAPI
-@XmlRootElement
 public class RMInitialState implements Serializable {
 
+    public static final Long EMPTY_STATE = -1L;
+
     private static final Logger LOGGER = Logger.getLogger(RMInitialState.class);
 
-    /**
-     * Nodes events
-     */
-    private Map<String, RMNodeEvent> nodeEvents = new ConcurrentHashMap<>();
-
-    /**
-     * Nodes sources AO living in RM
-     */
-    private Map<String, RMNodeSourceEvent> nodeSourceEvents = new ConcurrentHashMap<>();
+    private SortedUniqueSet<RMEvent> events = new SortedUniqueSet<>();
 
     /**
      * keeps track of the latest (biggest) counter among the 'nodeEvents' and 'nodeSourceEvents'
      */
     private AtomicLong latestCounter = new AtomicLong(0);
 
-    /**
-     * ProActive empty constructor
-     */
-    public RMInitialState() {
-
-    }
-
-    /**
-     * Creates an InitialState object.
-     *
-     * @param nodesEventList  RM's node events.
-     * @param nodeSourcesList RM's node sources list.
-     */
-    public RMInitialState(Map<String, RMNodeEvent> nodesEventList, Map<String, RMNodeSourceEvent> nodeSourcesList) {
-        this.nodeEvents = nodesEventList;
-        this.nodeSourceEvents = nodeSourcesList;
+    public void addAll(Collection<? extends RMEvent> toAdd) {
+        toAdd.forEach(rmEvent -> events.add(rmEvent));
+        latestCounter.set(Math.max(latestCounter.get(), findLargestCounter(events.getSortedItems())));
     }
 
     /**
      * Current version of RM portal and maybe other clients expects "nodesEvents" inside JSON
      *
-     * @return
+     * @return list of RMNodeEvent
      */
     public List<RMNodeEvent> getNodesEvents() {
-        return Collections.unmodifiableList(new ArrayList<>(this.nodeEvents.values()));
+        return getNodesEvents(events.getSortedItems());
     }
 
     /**
      * Current version of RM portal and maybe other clients expects "nodeSource" inside JSON
      *
-     * @return
+     * @return list of RMNodeSourceEvent
      */
     public List<RMNodeSourceEvent> getNodeSource() {
-        return Collections.unmodifiableList(new ArrayList<>(this.nodeSourceEvents.values()));
+        return getNodeSource(events.getSortedItems());
+    }
+
+    public List<RMNodeEvent> getNodesEvents(Collection<RMEvent> rmEvents) {
+        return rmEvents.stream()
+                       .filter(event -> event instanceof RMNodeEvent)
+                       .map(event -> (RMNodeEvent) event)
+                       .collect(Collectors.toList());
+    }
+
+    public List<RMNodeSourceEvent> getNodeSource(Collection<RMEvent> rmevents) {
```

Comment:
Can we rename this method `getNodeSourceEvents` or `getNodeSourcesEvents` ?

Reply:
Yes, this I can rename